torch/_dynamo/optimizations/analysis.py

```diff
@@ -122,9 +122,27 @@ def has_mutation(gm, example_inputs, inputs_only=False):
     # TODO - moco gives bad accuracy with Aliasing. gm is getting mutated in a bad way.
 
     if fake_tensors_available and config.fake_tensor_propagation:
-        with FakeTensorMode() as fake_mode:
+
+        def _wrap_to_fake_tensor(t, *, f_mode):
+            if type(t) in (torch.Tensor, torch.nn.Parameter):
+                static_shapes_ = config.dynamic_shapes is False
+                return fake_mode.from_tensor(
+                    t, static_shapes=config.dynamic_shapes is not False
+                )
+            else:
+                return t
+
+        # Our analysis pass should use dynamic shape tensor inputs
+        # when dynamic shapes are enabled.
+        # We don't actually care about the guards that are created
+        # on those shapes though, so just create a fresh ShapeEnv here.
+        from torch.fx.experimental.symbolic_shapes import ShapeEnv
+
+        with FakeTensorMode(
+            shape_env=ShapeEnv() if config.dynamic_shapes else None
+        ) as fake_mode:
             pass
```

Comment:
You don't need to do this anymore, it's not necessary to "pre-run" a mode before passing it on.

Reply:
Ditto - @bdhirsh ^

Reply:
cool - I cargo-culted the pre-run bit, we can def get rid of it.

Reply:
sg, will pull

Reply:
Done